### PR TITLE
28 create photo app

### PR DESCRIPTION
**Pull request by oshtohryn**
## TRELLO

* [Main Trello ticket](https://trello.com/c/bC4g4S1z/28-create-photo-app)

## Code reviewers

- [x] @KarnaushenkoVO
- [x] @PanLeon
- [x] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/photo/views.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        if not photo:
+            return HttpResponse(status=404)
+        update_data = json.loads(request.body.decode('utf-8'))
+        photo.update(**update_data)
+        return HttpResponse(status=200)
+
+    def delete(self, request, photo_id):
+        """DELETE request handler."""
+        photo = Photo.get_by_id(photo_id)
+        photo.delete()
```
Comment (KarnaushenkoVO) [inline myTrip/photo/views.py]: There is no handler for case, when method 'get_by_id' returns 'None'.

File: myTrip/photo/apps.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""."""
```
Comment (lhalam) [inline myTrip/photo/apps.py]: delete this line
pylint ignore=CVS, migrations, apps.py, urls.py, wsgi.py. local_setings.py, manage.py

File: myTrip/photo/apps.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+from django.apps import AppConfig
+
+
+class PhotoConfig(AppConfig):
+    """."""
```
Comment (lhalam) [inline myTrip/photo/apps.py]: delete this line

File: myTrip/photo/models.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    @staticmethod
+    def get_by_id(photo_id):
+        """Method return one element by id or None if exception."""
+        try:
+            return Photo.objects.get(id=photo_id)
+        except ObjectDoesNotExist:
+            return None
+
+    def create(self, src, user_id, trip_id=None, checkpoint_id=None, description=None):
```
Comment (lhalam) [inline myTrip/photo/models.py]: use static method

File: myTrip/photo/models.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        self.save()
+
+    def update(self, src, user_id):
+        """Updating photo model."""
+        self.src = src
+        self.user_id = user_id
+        self.save()
+
+    def to_dict(self):
+        """Convert queryset to dictionary."""
```
Comment (lhalam) [inline myTrip/photo/models.py]: for example:
```
"""Convert model object to dictionary.
        Return:
            dict:
                {
                    'id': id,
                    'user_id': name,
                    ...
                }
        """
```

File: myTrip/photo/models.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+     :argument description: str - description to photo
+    ."""
+    src = models.URLField(max_length=200)
+    user_id = models.IntegerField()
+    trip_id = models.IntegerField(null=True)
+    checkpoint_id = models.IntegerField(null=True)
+    description = models.TextField(null=True)
+
+    @staticmethod
+    def get_by_id(photo_id):
```
Comment (lhalam) [inline myTrip/photo/models.py:25]: for example:
```
"""
Get Like with given trip id
 Args:
    trip_id (int): trip id.
Returns:
    QuerySet<Like>: QuerySet of Like.
"""
```

**Review by yuriiZherebukh — APPROVED** [review]

**Pull request merged**